core/src/main/java/org/apache/shiro/realm/activedirectory/ActiveDirectoryRealm.java

```diff
@@ -105,7 +105,12 @@ protected AuthenticationInfo queryForAuthenticationInfo(AuthenticationToken toke
         // Binds using the username and password provided by the user.
         LdapContext ctx = null;
         try {
-            ctx = ldapContextFactory.getLdapContext(upToken.getUsername(), String.valueOf(upToken.getPassword()));
+            String userPrincipalName = upToken.getUsername();
+            if (this.principalSuffix != null) {
+                userPrincipalName = upToken.getUsername() + this.principalSuffix;
+            }
```

Comment:
This _should_ be already happening here: https://github.com/apache/shiro/blob/master/core/src/main/java/org/apache/shiro/realm/ldap/DefaultLdapContextFactory.java#L221-L222

Is this not happening in your setup ?